### DISCUSSION TITLE
Hide Frida Package and Process name

### DISCUSSION THREAD

**Discussion by paul-tharun** [frida/frida#2416] [Q&A]
I am trying to bypass root detection and reverse engineer the communication protocol of an app.
But the app is uses checks for common locations for root apps so it ends up detecting the frida shared object file that is created on frida start in \data\local\tmp directory and crashes on start. 

So, I am trying to change the name of the frida process and the file name to bypass this.

Can some one point me some resources for it please? I tried :

1. To recomplie the app after changing all occurrences of the string frida to something else. I am not able to compile as the process errors out.
2. Tried to change all occurences of the string 'frida' in the complied frida-server file. The device restarts when I the run it.

**Comment by shinopayment** [comment]
hi did u able to find out the way to prevent app detected the frida service

**Reply by paul-tharun** [reply]
Hey, I wasn't able to figure it and had to move on to different projects so this is still open
